src/main/java/fr/sncf/osrd/train/TrainState.java

```diff
@@ -197,35 +196,56 @@ private void step(
                 throw new SimulationError("train physics numerical integration doesn't seem to stop");
             var distanceStep = goalPathPosition - location.getPathPosition();
             step(locationChange, 1.0, distanceStep);
+            // Stop the evolution if the train has stopped
+            if (speed < 0.0000001)
+                break;
         }
 
         return locationChange;
     }
 
+    /**  Create a location change from the current state to current simulation time */
+    public Train.TrainStateChange evolveStateUntilNow(Simulation sim) {
+        var locationChange = new Train.TrainStateChange(sim, trainSchedule.trainID, this);
+
+        while (this.time + 1.0 < sim.getTime())
```

Comment:
Where does the value `1.0` come from?

Reply:
For the moment we use a fixed time step of 1 second. Ideally we should have a variable time step.